modules/i18n/server/controllers/backend.server.controller.js

```diff
@@ -0,0 +1,54 @@
+const path = require('path');
+const fs = require('fs-extra');
+
+/**
+ * We want to process only one unsaved translation at the time.
+ * Otherwise the output files will be inconsistent.
+ * We make a promise chain: The next promise will start executing after the previous one finished
+ */
+const queue = { promise: Promise.resolve() };
```

Comment:
I've been using [`async.queue`](https://caolan.github.io/async/v2/docs.html#queue) helper for these but delighted to see how few lines custom solution is. :-)